File: C++Solutions/FindDisappearedNumbers.cpp
#include <iostream>
#include <set>
#include <vector>
using namespace std;

class Solution
{
public:
    vector<int> findDisappearedNumbers(vector<int>& nums) {
        vector<int> missingNums; //variable to store missing numbers
        int maxNum = nums.size(); //find length
        set<int> numberSet(nums.begin(), nums.end()); //create set

        //loop through 1 > length; grab missing numbers not in set
        for(int i = 1; i <= maxNum; i++){
            if (numberSet.count(i) == 0){ //cant use .contains before C++20
                missingNums.push_back(i);
            }
        }
        //return variable array of missing set 
        return missingNums;
    }
};

int main() {
    Solution solution;
    vector<int> nums = {4, 3, 2, 7, 8, 2, 3, 1};

    vector<int> result = solution.findDisappearedNumbers(nums);

    // Print the missing numbers
    cout << "Missing numbers: ";
    for (int num : result) {
        cout << num << " ";
    }

    return 0;
}

File: PythonSolutions/FindDisappearedNumbers.py
from typing import List

class Solution:
    def findDisappearedNumbers(self, nums: List[int]) -> List[int]:

        missing_nums = []  # Variable to store missing numbers
        max_num = len(nums)  # Find length
        number_set = set(nums)  # Create set from nums

        # Loop through 1 to length; grab missing numbers not in the set
        for i in range(1, max_num + 1):
            if i not in number_set:  # Check if i is not in the set
                missing_nums.append(i)  # Append missing number to the list

        # Return variable array of missing numbers 
        return (missing_nums)
